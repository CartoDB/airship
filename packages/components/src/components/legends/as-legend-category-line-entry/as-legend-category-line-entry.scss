@import '../../../../../styles/src/core/_dev';

as-legend-category-line-entry {
  .as-legend-category-line-entry--wrapper {
    display: flex;
    align-items: center;
  }

  .as-legend-category-line-entry--line {
    width: get-spacing(4);
    height: 0;
    margin: auto;
  }
}
