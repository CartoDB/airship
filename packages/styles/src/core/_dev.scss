@import 'tools/_mixins';
@import 'variables/_variables';
