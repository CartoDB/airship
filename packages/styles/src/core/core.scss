@import 'variables/_variables';

* {
  box-sizing: border-box;
  font-family: sans-serif;
}

:root {
  --primary-color: #{$primary-color};
}
