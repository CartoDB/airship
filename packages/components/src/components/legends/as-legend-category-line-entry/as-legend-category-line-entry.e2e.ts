import { E2EPage, newE2EPage } from '@stencil/core/testing';

describe('as-legend-category-line-entry', async () => {
  describe('Rendering', async () => {
    let page: E2EPage;

    it('should render label and color, with 1.5px solid style by default', async () => {
      page = await newE2EPage({ html: `
        <as-legend-category-line-entry>
        </as-legend-category-line-entry>
      ` });

      const element = await page.find('as-legend-category-line-entry');
      element.setProperty('label', 'Hello there');
      element.setProperty('color', '#FABADA');
      await page.waitForChanges();
      expect(element.outerHTML).toMatchSnapshot();
    });

    it('should support custom stroke width', async () => {
      page = await newE2EPage({ html: `
        <as-legend-category-line-entry>
        </as-legend-category-line-entry>
      ` });

      const element = await page.find('as-legend-category-line-entry');
      element.setProperty('label', 'Hello there');
      element.setProperty('color', '#FABADA');
      element.setProperty('width', '10');
      await page.waitForChanges();
      expect(element.outerHTML).toMatchSnapshot();
    });

    it('should support custom stroke style', async () => {
      page = await newE2EPage({ html: `
        <as-legend-category-line-entry>
        </as-legend-category-line-entry>
      ` });

      const element = await page.find('as-legend-category-line-entry');
      element.setProperty('label', 'Hello there');
      element.setProperty('color', '#FABADA');
      element.setProperty('strokeStyle', 'dotted');
      await page.waitForChanges();
      expect(element.outerHTML).toMatchSnapshot();
    });
  });
});
