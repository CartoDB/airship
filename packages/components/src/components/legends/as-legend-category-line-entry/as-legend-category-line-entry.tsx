import { Component, Prop, State, Watch } from '@stencil/core';

const MAX_WIDTH = 16;

@Component({
  shadow: false,
  styleUrl: './as-legend-category-line-entry.scss',
  tag: 'as-legend-category-line-entry',
})
export class LegendCategoryLineEntry {
  @Prop() public label: string;
  @Prop() public color: string;
  @Prop() public strokeStyle: string = 'solid';
  @Prop() public width: number = 1.5;

  @State() private _width: number;

  public componentWillLoad() {
    this._widthChanged(this.width);
  }

  public render() {
    return (
      <div class='as-legend-category-line-entry--wrapper'>
        <div class='as-legend-category--figure--wrapper'>
          <div class='as-legend-category-line-entry--line' style={this.getStyle()}></div>
        </div>
        <span class='as-legend-category--label'>{this.label}</span>
      </div>
    );
  }

  @Watch('width')
  public _widthChanged(newValue: number) {
    this._width = Math.min(MAX_WIDTH, newValue);
  }

  private getStyle() {
    return {
      borderTopColor: this.color,
      borderTopStyle: this.strokeStyle,
      borderTopWidth: `${this._width}`
    };
  }
}
