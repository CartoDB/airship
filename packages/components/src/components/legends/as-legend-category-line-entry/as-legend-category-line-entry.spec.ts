import { newSpecPage, SpecPage } from '@stencil/core/testing';
import { LegendCategoryLineEntry } from './as-legend-category-line-entry';

describe('as-legend-category-line-entry', () => {
  describe('Rendering', () => {
    let page: SpecPage;

    it('should render label and color, with 1.5px solid style by default', async () => {
      page = await newSpecPage({
        components: [LegendCategoryLineEntry],
        html: `<as-legend-category-line-entry></as-legend-category-line-entry>`
      });

      const element = page.root;
      element.label = 'Hello there';
      element.color = '#FABADA';
      await page.waitForChanges();
      expect(element.outerHTML).toMatchSnapshot();
    });

    it('should support custom stroke width', async () => {
      page = await newSpecPage({
        components: [LegendCategoryLineEntry],
        html: `<as-legend-category-line-entry></as-legend-category-line-entry>`
      });

      const element = page.root;
      element.label = 'Hello there';
      element.color = '#FABADA';
      element.width = '10';
      await page.waitForChanges();
      expect(element.outerHTML).toMatchSnapshot();
    });

    it('should support custom stroke style', async () => {
      page = await newSpecPage({
        components: [LegendCategoryLineEntry],
        html: `<as-legend-category-line-entry></as-legend-category-line-entry>`
      });

      const element = page.root;
      element.label = 'Hello there';
      element.color = '#FABADA';
      element.strokeStyle = 'dotted';
      await page.waitForChanges();
      expect(element.outerHTML).toMatchSnapshot();
    });
  });
});
