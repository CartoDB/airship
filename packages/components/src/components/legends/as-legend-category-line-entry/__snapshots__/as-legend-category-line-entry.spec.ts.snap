// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`as-legend-category-line-entry Rendering should render label and color, with 1.5px solid style by default 1`] = `"<as-legend-category-line-entry><div class=\\"as-legend-category-line-entry--wrapper\\"><div class=\\"as-legend-category--figure--wrapper\\"><div class=\\"as-legend-category-line-entry--line\\" style=\\"border-top-style: solid; border-top-width: 1.5; border-top-color: #FABADA;\\"></div></div><span class=\\"as-legend-category--label\\">Hello there</span></div></as-legend-category-line-entry>"`;

exports[`as-legend-category-line-entry Rendering should support custom stroke style 1`] = `"<as-legend-category-line-entry><div class=\\"as-legend-category-line-entry--wrapper\\"><div class=\\"as-legend-category--figure--wrapper\\"><div class=\\"as-legend-category-line-entry--line\\" style=\\"border-top-style: dotted; border-top-width: 1.5; border-top-color: #FABADA;\\"></div></div><span class=\\"as-legend-category--label\\">Hello there</span></div></as-legend-category-line-entry>"`;

exports[`as-legend-category-line-entry Rendering should support custom stroke width 1`] = `"<as-legend-category-line-entry><div class=\\"as-legend-category-line-entry--wrapper\\"><div class=\\"as-legend-category--figure--wrapper\\"><div class=\\"as-legend-category-line-entry--line\\" style=\\"border-top-style: solid; border-top-width: 10; border-top-color: #FABADA;\\"></div></div><span class=\\"as-legend-category--label\\">Hello there</span></div></as-legend-category-line-entry>"`;
